uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -618,7 +628,10 @@ define([
      * Sets day selection mode: multiple days or a single day
      */
     function setDaySelectionMode () {
-      if ((vm.isMode('edit') && vm.request.from_date === vm.request.to_date) ||
+      if (
+        ((vm.isMode('edit') || vm.isMode('view')) &&
+        _convertDateToServerFormat(vm.request.from_date) ===
+        _convertDateToServerFormat(vm.request.to_date)) ||
         (vm.isMode('create') && vm.isLeaveType('sickness'))) {
```

Comment:
Here the the condition has been complex. Can we separate them into separate functions?